File: temp/comprehensive_fix.js
const fs = require('fs');
const path = require('path');

const dir = "D:/新课开发/工作手册/高潜员工加速成长路径设计/完整课程包/02-授课PPT/slides";

// Get all slide files
const files = fs.readdirSync(dir).filter(f => f.startsWith('slide-') && f.endsWith('.js'));

files.forEach(f => {
  const fp = path.join(dir, f);
  let content = fs.readFileSync(fp, 'utf8');
  
  // Fix 1: Remove backslash escaping of quotes - replace \' with ' and \" with "
  content = content.replace(/\'/g, "'");
  content = content.replace(/\\"/g, '"');
  
  // Fix 2: For addText calls with Chinese text that has embedded quotes,
  // find and fix patterns like addText('text with 'quotes' text',
  const lines = content.split('\n');
  const fixed = lines.map(line => {
    // Match addText with single-quoted string containing single quotes
    // Pattern: addText('...anything with 'quoted' parts...',
    if (line.match(/addText\('[^']*'[^']*'\s*,/)) {
      // Change to double quotes
      line = line.replace(/addText\('/, 'addText("').replace(/'\s*,/, '",');
    }
    return line;
  });
  content = fixed.join('\n');
  
  fs.writeFileSync(fp, content);
});

console.log(`Fixed ${files.length} files`);

File: temp/comprehensive_fix.py
#!/usr/bin/env python3
"""Comprehensive fix for all slide JS files"""
import re, os

base = r"D:\新课开发\数字化转型\4.价值密度战略设计：从交付产品到交付结果\授课PPT\slides"

def read(path):
    with open(path, 'r', encoding='utf-8') as f:
        return f.read()

def write(path, content):
    with open(path, 'w', encoding='utf-8') as f:
        f.write(content)

def fix_file(path):
    content = read(path)
    original = content

    # Fix 1: Module-level theme reference (e.g., `theme` in slideConfig outside function)
    content = re.sub(r'const slideConfig = \{[^}]*theme\s*\};?\n', '', content)

    # Fix 2: Arrays ending with single quote: ["a", "b', ...] -> ["a", "b", ...]
    # Pattern: ...item'" followed by ] or , ]
    lines = content.split('\n')
    fixed_lines = []
    for line in lines:
        # Look for array items ending with single quote before ] or ,
        if re.search(r'"\',?\s*\]|"\',?\s*,', line):
            # Replace trailing single quote in array items
            line = re.sub(r'"(\s*[,\]])', r'"\1', line)
        fixed_lines.append(line)
    content = '\n'.join(fixed_lines)

    # Fix 3: Strings with nested quotes
    def fix_nested(m):
        prefix = m.group(1)
        outer = m.group(2)
        inner = m.group(3)
        suffix = m.group(4)
        inner_double = inner.count('"')
        inner_single = inner.count("'")
        if inner_double > 0 and inner_single == 0:
            if outer == '"':
                return prefix + "'" + inner + "'" + suffix
        elif inner_single > 0 and inner_double == 0:
            if outer == "'":
                return prefix + '"' + inner + '"' + suffix
        return m.group(0)

    content = re.sub(
        r'(slide\.addText|slide\.addShape)\(((?:\"(?:[^\"\\]|\\.)*\"|\'(?:[^\'\\]|\\.)*\')|),(.*?)(,(?:\s*\{[^}]*\})?)\)',
        fix_nested,
        content,
        flags=re.DOTALL
    )

    # Fix 4: Wrong API (pptx.ShapeType.rect)
    content = content.replace('pptx.ShapeType.rect', 'pres.shapes.RECTANGLE')
    content = content.replace('pptx.ShapeType.ellipse', 'pres.shapes.OVAL')
    content = content.replace('pptx.ShapeType.', 'pres.shapes.')

    # Fix 5: return pptx instead of slide
    content = content.replace('return pptx;', 'return slide;')

    return content if content != original else None

files = sorted([f for f in os.listdir(base) if f.startswith('slide-') and f.endswith('.js')],
               key=lambda x: int(re.search(r'\d+', x).group()))

fixed_count = 0
for fname in files:
    path = os.path.join(base, fname)
    try:
        result = fix_file(path)
        if result:
            write(path, result)
            print(f"Fixed: {fname}")
            fixed_count += 1
    except Exception as e:
        print(f"Error {fname}: {e}")

print(f"\nTotal fixed: {fixed_count}")
